app.py

```diff
@@ -0,0 +1,61 @@
+from flask import Flask, render_template, redirect, url_for, request, session, flash
+from functools import wraps
+
+app = Flask(__name__)
+
+app.secret_key = "my precious" 
+
+def login_required(f):
+    @wraps(f)
+    def wrap(*args, **kwargs):
+        if 'logged_in' in session:
+            return f(*args, **kwargs)
+        else:
+            flash('You need to login first.')
+            return redirect(url_for('login'))
+    return wrap
+
+@app.route('/login', methods=['GET', 'POST'])
+def login():
+	error = None
+	if request.method == 'POST':
+		if request.form['username'] != 'admin' or request.form['password'] != 'admin':
```

Comment:
So we are only letting someone into the system if they pass in username `admin` and password `admin`?

Reply:
For now, yes. As I get further into development will add in the ability to create a unique username and password for each user

Reply:
👍